### PR TITLE
chore: clean up some useless scripts

### PR DESCRIPTION
**Pull request by whynowy**
### Modifications

- Clean up some useless scripts such as `make install`, `make uninstall`, `make undeploy`.
- Update `make deploy` to `make start` to make it consistent with Numaflow.
- Remove `kustomize` dependency, use `kubectl kustomize` instead.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -152,22 +152,10 @@ ifndef ignore-not-found
   ignore-not-found = false
 endif
 
-.PHONY: install
-install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
```
Comment (whynowy) [inline Makefile:156]: We won't do CRD installation only most of the time.

File: Makefile [hunk 2]
```diff
@@ -152,22 +152,10 @@ ifndef ignore-not-found
...
-	$(KUSTOMIZE) build config/crd | $(KUBECTL) apply -f -
-
-.PHONY: uninstall
-uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
```
Comment (whynowy) [inline Makefile:160]: Can no do CRD uninstallation only if there are already CR objects created.

File: Makefile [hunk 3]
```diff
@@ -152,22 +152,10 @@ ifndef ignore-not-found
...
-	$(KUSTOMIZE) build config/crd | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
-
-.PHONY: deploy
-deploy: docker-build manifests
+.PHONY: start
```
Comment (whynowy) [inline Makefile:155]: Update `deploy` to `start` to make it consistent with Numaflow.

**Comment by juliev0** [comment]
can you add information to "Modifications" and "Verification" in PR Description?

File: Makefile [hunk 4]
```diff
@@ -152,22 +152,10 @@ ifndef ignore-not-found
...
+start: docker-build manifests
 	$(KUBECTL) kustomize tests/manifests | sed 's/CLUSTER_NAME_VALUE/$(CLUSTER_NAME)/g' | sed 's@quay.io/numaproj/@$(IMAGE_NAMESPACE)/@' | sed 's/$(IMG):$(BASE_VERSION)/$(IMG):$(VERSION)/' | $(KUBECTL) apply -f -
 
-.PHONY: undeploy
```
Comment (whynowy) [inline Makefile:167]: Do not need it as we can not only uninstall CRD and controller only if there are already CR objects existing.

File: Makefile [hunk 5]
```diff
@@ -177,23 +165,12 @@ $(LOCALBIN):
...
-KUSTOMIZE ?= $(LOCALBIN)/kustomize
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v5.2.1
 CONTROLLER_TOOLS_VERSION ?= v0.13.0
 
-.PHONY: kustomize
-kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary. If wrong version is installed, it will be removed before downloading.
```
Comment (whynowy) [inline Makefile:189]: Use `kubectl kustomize` instead.

**Review by xdevxy — APPROVED** [review]

**Pull request merged**